src/journal-remote/journal-upload.c

```diff
@@ -681,6 +756,21 @@ static int parse_argv(int argc, char *argv[]) {
                                 return log_error_errno(r, "Failed to add paths: %m");
                         break;
 
+                case ARG_HEADERS:
+                        r = strv_headers_replace_consume(&arg_headers, optarg);
+                        if (r < 0)
+                                return log_error_errno(
+                                                r,
+                                                "--headers value should contain '='-delimited pairs, separated by comma");
+                        break;
+
+                case ARG_COMPRESS:
+                        if (!compression_supported(COMPRESSION_ZSTD))
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                                       "ZSTD compression is not supported");
+                        arg_compress = true;
```

Comment:
so, even if only zstd is supported initially: the parameter should be generic, i.e. probably be specified as --compress=zstd, to keep the door open for adding more later on